packages/devtools_app/lib/src/screens/inspector/inspector_data_models.dart

```diff
@@ -439,62 +454,62 @@ class FlexLayoutProperties extends LayoutProperties {
   }
 
   Axis get crossAxisDirection {
-    return direction == Axis.horizontal ? Axis.vertical : Axis.horizontal;
+    return direction;
   }
 
-  double get mainAxisDimension => dimension(direction);
+  double? get mainAxisDimension => dimension(direction);
 
-  double get crossAxisDimension => dimension(crossAxisDirection);
+  double? get crossAxisDimension => dimension(crossAxisDirection);
 
   @override
   bool get isOverflowWidth {
     if (direction == Axis.horizontal) {
-      return width + overflowEpsilon < sum(childrenWidths);
+      return width! + overflowEpsilon < sum(childrenWidths!.cast<double>());
     }
-    return width + overflowEpsilon < max(childrenWidths);
+    return width! + overflowEpsilon < max(childrenWidths!.cast<double>());
   }
 
   @override
   bool get isOverflowHeight {
     if (direction == Axis.vertical) {
-      return height + overflowEpsilon < sum(childrenHeights);
+      return height! + overflowEpsilon < sum(childrenHeights!.cast<double>());
     }
-    return height + overflowEpsilon < max(childrenHeights);
+    return height! + overflowEpsilon < max(childrenHeights!.cast<double>());
   }
 
   bool get startIsTopLeft {
-    assert(direction != null);
     switch (direction) {
       case Axis.horizontal:
         switch (textDirection) {
           case TextDirection.ltr:
+          case null:
             return true;
           case TextDirection.rtl:
             return false;
         }
-        break;
       case Axis.vertical:
         switch (verticalDirection) {
           case VerticalDirection.down:
+          case null:
             return true;
           case VerticalDirection.up:
             return false;
         }
     }
-    return true;
   }
 
   /// render properties for laying out rendered Flex & Flex children widgets
   /// the computation is similar to [RenderFlex].performLayout() method
   List<RenderProperties> childrenRenderProperties({
-    @required double smallestRenderWidth,
-    @required double largestRenderWidth,
-    @required double smallestRenderHeight,
-    @required double largestRenderHeight,
-    @required double Function(Axis) maxSizeAvailable,
+    required double? smallestRenderWidth,
+    required double? largestRenderWidth,
+    required double? smallestRenderHeight,
+    required double? largestRenderHeight,
+    required double Function(Axis) maxSizeAvailable,
   }) {
     /// calculate the render empty spaces
-    final freeSpace = dimension(direction) - sum(childrenDimensions(direction));
+    final freeSpace = dimension(direction)! -
+        sum(childrenDimensions(direction).cast<double>());
```

Comment:
this cast shouldn't be necessary once childrenDimensions is changed to return `List<double>`